File: code/gms-frontend/src/app/common/service/event-service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Event } from "../model/event.model";
import { EventList } from "../model/event-list.model";
import { ServiceBase } from "./service-base";

/**
 * @author Peter Szrnka
 */
@Injectable()
export class EventService extends ServiceBase<Event, EventList> {

    constructor(http : HttpClient) {
        super(http, "event");
    }
}

File: code/gms-backend/src/main/java/io/github/gms/functions/event/EventService.java
package io.github.gms.functions.event;

import io.github.gms.common.abstraction.BatchDeletionService;
import io.github.gms.common.abstraction.GmsService;
import io.github.gms.common.dto.IntegerValueDto;
import io.github.gms.common.model.UserEvent;
import org.springframework.data.domain.Pageable;

/**
 * @author Peter Szrnka
 * @since 1.0
 */
public interface EventService extends GmsService, BatchDeletionService {

	void saveUserEvent(UserEvent event);
	
	EventListDto list(Pageable pageable);
	
	EventListDto listByUser(Long userId, Pageable pageable);

    IntegerValueDto getUnprocessedEventsCount();
}
